huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go

```diff
@@ -293,18 +290,119 @@ func buildCreateDscInstanceRequestBodyProductInfos(d *schema.ResourceData) []map
 		if size, ok := d.GetOk("database_expansion_package"); ok {
 			rst = append(rst, map[string]interface{}{
 				"cloudServiceType":      cloudServiceType,
-				"productId":             dscDBProductIdPro,
 				"resourceType":          resourceTypeDB,
 				"resourceSpecCode":      resourceSpecCodeProDB,
 				"resourceSize":          utils.ValueIngoreEmpty(size),
 				"resourceSizeMeasureId": resourceSizeMeasureIdDB,
 			})
 		}
 	}
+	if err := addProductIdToProductInfo(d, cfg, rst); err != nil {
+		return nil, err
+	}
+
+	return rst, nil
+}
 
+func addProductIdToProductInfo(d *schema.ResourceData, cfg *config.Config, rst []map[string]interface{}) error {
+	region := cfg.GetRegion(d)
+	productInfos := make([]map[string]interface{}, len(rst))
+	period := d.Get("period").(int)
+	periodUnit := d.Get("period_unit").(string)
+	for i, product := range rst {
+		productInfos[i] = buildProductInfos(cfg.GetRegion(d), periodUnit, strconv.Itoa(i+1), period, product)
+	}
+	products, err := getOrderProductId(cfg, region, productInfos)
+	if err != nil {
+		return fmt.Errorf("error getting DSC order product id infos: %s", err)
```

Comment:
```suggestion
		return fmt.Errorf("error getting DSC order product ID infos: %s", err)
```
Please check the other parts.